pandas/formats/format.py

```diff
@@ -903,8 +916,46 @@ def get_col_type(dtype):
                          if x else '{}') for x in row]
             else:
                 crow = [x if x else '{}' for x in row]
+            if i < nlevels and self.fmt.header and self.multicol:
```

Comment:
This might be easier to implement using something like what's in https://github.com/pydata/pandas/blob/master/pandas/formats/style.py, but it'd be some work to refactor using that.